server/src/com/thoughtworks/go/config/MergedGoConfig.java

```diff
@@ -101,7 +102,7 @@ public void tryAssembleMergedConfig(GoConfigHolder cruiseConfigHolder,List<Parti
             // save to cache and fire event
             this.saveValidConfigToCacheAndNotifyConfigChangeListeners(newConfigHolder);
         } catch (Exception e) {
-            LOGGER.error(String.format("Failed validation of merged configuration: %s", e));
+            LOGGER.error("Failed validation of merged configuration: {}", e);
```

Comment:
It won't fail. But, it's not ideal. I'm changing it. Here's what happens:

```java
@Test
public void shouldLogException() throws Exception {
    Logger LOGGER = LoggerFactory.getLogger(ApprovalTest.class);
    RuntimeException e = new RuntimeException("Hello");
    LOGGER.error("=== Message 1: {} - {} _-_", e.getClass(), e.getMessage());
    LOGGER.error("--- Message 2: {} +++", e);
}
```

```
<=== Message 1: class java.lang.RuntimeException - Hello _-_>

<--- Message 2: {} +++>
java.lang.RuntimeException: Hello
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:606)
  at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
  ...
```

As you can see, slf4j will just print extra `{}` arguments, if there's nothing which matches and the exception shows fully. The earlier message seemed to just want to call `e.toString()` and not show the stack trace. I'll just do that too.